numpy/_core/src/multiarray/multiarraymodule.c

```diff
@@ -5129,7 +5163,7 @@ PyMODINIT_FUNC PyInit__multiarray_umath(void) {
 
     // initialize static reference to a zero-like array
     npy_static_pydata.zero_pyint_like_arr = PyArray_ZEROS(
-            0, NULL, NPY_LONG, NPY_FALSE);
+            0, NULL, NPY_DEFAULT_INT, NPY_FALSE);
```

Comment:
good catch, hope there isn't a long tail of these

Reply:
Hopefully not (but I doubt it).  This one actually had no effect at all previously, just now and even then only in cast-safety subtleties.